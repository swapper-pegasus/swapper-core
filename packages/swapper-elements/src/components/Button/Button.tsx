import React from 'react'

type Props = {
  onClick: () => void;
  children?: JSX.Element | string;
  disabled?: boolean,
  withGradient?: boolean,
  dataTestId?: string,
  type?: string,
};

export function Button ({
  onClick,
  children,
  disabled,
  withGradient,
  dataTestId,
  type
}: Props) {
  const style = 'disabled:cursor-not-allowed disabled:opacity-20 bg-primary rounded-full hover:bg-primary-dark text-white text-sm px-8 py-2'
  const styleGradient = 'bg-gradient-to-r from-primary-darkest to-primary-lightest hover:bg-blue-500'
  const styleButtonLink = 'text-primary underline'

  const finalStyle = () => {
    if (type === 'link') {
      return styleButtonLink
    }
    return withGradient ? `${style} ${styleGradient}` : style
  }

  return (
      <button
        className={finalStyle()}
        onClick={onClick}
        disabled={disabled}
        data-testid={dataTestId}
      >
        {children}
      </button>
  )
}
