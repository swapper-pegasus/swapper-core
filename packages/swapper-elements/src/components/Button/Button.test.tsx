import React from 'react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import { Button } from './Button'

describe('Button', () => {
  it('Render default', () => {
    const mockFn = jest.fn()
    const tree = renderer.create(<Button onClick={mockFn}><p>children</p></Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Render link', () => {
    const mockFn = jest.fn()
    const tree = renderer.create(<Button type='link' onClick={mockFn}><p>children</p></Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Render withGradient', () => {
    const mockFn = jest.fn()
    const tree = renderer.create(<Button withGradient onClick={mockFn}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Render disabled', () => {
    const mockFn = jest.fn()
    const tree = renderer.create(<Button disabled onClick={mockFn}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
