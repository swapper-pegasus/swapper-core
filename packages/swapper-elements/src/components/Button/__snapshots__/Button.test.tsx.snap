// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render default 1`] = `
<button
  className="disabled:cursor-not-allowed disabled:opacity-20 bg-primary rounded-full hover:bg-primary-dark text-white text-sm px-8 py-2"
  onClick={[MockFunction]}
>
  <p>
    children
  </p>
</button>
`;

exports[`Button Render disabled 1`] = `
<button
  className="disabled:cursor-not-allowed disabled:opacity-20 bg-primary rounded-full hover:bg-primary-dark text-white text-sm px-8 py-2"
  disabled={true}
  onClick={[MockFunction]}
/>
`;

exports[`Button Render link 1`] = `
<button
  className="text-primary underline"
  onClick={[MockFunction]}
>
  <p>
    children
  </p>
</button>
`;

exports[`Button Render withGradient 1`] = `
<button
  className="disabled:cursor-not-allowed disabled:opacity-20 bg-primary rounded-full hover:bg-primary-dark text-white text-sm px-8 py-2 bg-gradient-to-r from-primary-darkest to-primary-lightest hover:bg-blue-500"
  onClick={[MockFunction]}
/>
`;
